aiida/orm/nodes/data/code/containerized.py

```diff
@@ -0,0 +1,167 @@
+
+    @property
+    def engine_command(self) -> str:
+        """Return the engine command with image as template field of the containerized code
+
+        :return: The engine command of the containerized code
+        """
+        return self.base.attributes.get(self._KEY_ATTRIBUTE_ENGINE_COMMAND)
+
+    @engine_command.setter
+    def engine_command(self, value: str) -> None:
+        """Set the engine command of the containerized code
+
+        :param value: The engine command of the containerized code
+        """
+        type_check(value, str)
+
+        if '{image}' not in value:
+            raise ValueError("the '{image}' template field should be in engine command.")
+
+        self.base.attributes.set(self._KEY_ATTRIBUTE_ENGINE_COMMAND, value)
+
+    @property
+    def escape_exec_line(self) -> bool:
+        """True for escape whole execute line after engine command in double quotes.
+
+        :return: If the execute line after engine command is escaped in double quotes
+        """
+        return self.base.attributes.get(self._KEY_ATTRIBUTE_ESCAPE_EXEC_LINE)
+
+    @escape_exec_line.setter
+    def escape_exec_line(self, value: bool) -> None:
+        """Set whether to escape execute line in double quotes
+
+        :param value: True for escape execute line in double quotes, False for not escape.
+        """
+        type_check(value, bool)
+
+        self.base.attributes.set(self._KEY_ATTRIBUTE_ESCAPE_EXEC_LINE, value)
+
+    @property
+    def image(self) -> str:
+        """The image of container
+
+        :return: The image of container.
+        """
+        return self.base.attributes.get(self._KEY_ATTRIBUTE_IMAGE)
+
+    @image.setter
+    def image(self, value: str) -> None:
+        """Set the image name of container
+
+        :param value: The image name of container.
+        """
+        type_check(value, str)
+
+        self.base.attributes.set(self._KEY_ATTRIBUTE_IMAGE, value)
+
+    def get_engine_command(self) -> str:
+        """Return the engine command with image field set with image name.
+
+        :return: The engine command in script job with image field set with the image name.
+        """
+        cmdline = self.engine_command.format(image=self.image)
+
+        return cmdline.split()
+
+    @classmethod
+    def _get_cli_options(cls) -> dict:
+        """Return the CLI options that would allow to create an instance of this class."""
+        options = {
+            'engine_command': {
+                'required': True,
+                'prompt': 'Engine command',
+                'help': 'The command to run container must contain {image} for image.',
+                'type': click.STRING,
+            },
+            'image': {
+                'required': True,
+                'type': click.STRING,
+                'prompt': 'Image',
+                'help': 'Image of the container to run executable.',
+            },
+            'escape_exec_line': {
+                'type': click.BOOL,
```

Comment:
Better to use `is_flag` because then we get `--escape-exec-line/--no-escape-exec-line`. This works more nicely with a default that is `True`.
```suggestion
                'is_flag': True,
```